src/Widgets/WindowClone.vala

```diff
@@ -510,7 +530,17 @@ namespace Gala {
                     close_button.x = -close_button.width * 0.5f;
                     break;
             }
+
+            bool show = has_pointer && !in_slot_animation;
+            close_button.opacity = show ? 255 : 0;
+            window_title.opacity = show ? 255 : 0;
+
+            window_title.text = window.get_title ();
+            window_title.max_width = dest_width - (TITLE_MAX_WIDTH_MARGIN * scale_factor);
```

Comment:
These two sets alone will create the text actor, remove it and add it to the tooltip twice causing unnecessary work we can avoid. I would simply declare the `draw` method public or declare an `update` method that takes the text and max width and have it call draw once.

Reply:
This is optimized now.